proto/spacemesh/types.proto

```diff
@@ -109,59 +124,85 @@ message Transaction {
     TransactionId prev_atx = 11;  // previous ATX. used in atx.
 }
 
+message Transactions {
+    uint64 total_items = 1; // total server available items. Used to query with new offset
+    repeated Transaction transactions = 2;
+}
+
+message Reward {
+    uint64 layer = 1; // layer award was paid in
+    Amount total = 2; // total reward paid (sum of tx fee and layer reward)
+    Amount layer_reward = 3;
+    uint64 layer_computed = 4; // layer number of the layer when reward was computed
+    // tx_fee = total - layer_reward
+    AccountId coinbase = 5; // account awarded this reward
+    SmesherId smesher = 6;  // it will be nice to always have this in reward events
+}
+
+message Rewards {
+    uint64 total_items = 1; // total server available items. Used to query with new offset
+    repeated Reward rewards = 2;
+}
+
 message Block {
-    bytes id = 1;                           // block hash
+    bytes id = 1; // block hash
     repeated Transaction transactions = 2;
 }
 
 message Layer {
-    uint64 number = 1;              // layer number - not hash - layer content may change
     enum LayerStatus {
-        UNDEFINED = 0;
-        UNKNOWN = 1;
-        APPROVED = 2;
-        CONFIRMED = 3;
+        UNKNOWN = 0;
+        APPROVED = 1;
+        CONFIRMED = 2;
     }
+
+    uint64 number = 1;  // layer number - not hash - layer content may change
     LayerStatus status = 2;
     bytes hash = 3;                 // computer layer hash - do we need this?
     repeated Block blocks = 4;      // layer's blocks
     bytes root_state_hash = 5;      // when available - the root state hash of global state in this layer
 }
 
+// current node status
 message NodeStatus {
-    uint64 known_peers = 1;
-    uint64 min_peers = 2;
-    uint64 max_peers = 3;
-    bool   is_synced = 4;
-    uint64 synced_layer = 5;
-    uint64 current_layer = 6;
-    uint64 verified_layer = 7;
+    uint64 connected_peers = 1;       // number of connected neighbors
+    bool   is_synced = 2;             // Is meshed synced
+    uint64 synced_layer = 3;          // the last layer node has synced
+    uint64 top_layer = 4;             // top layer is the tip
+    uint64 verified_layer = 5;        // the last layer node has verified
+}
+
+// Indicates a change in node status
+message NodeStatusEvent {
+    enum NodeStatusEvent {
+      UNDEFINED = 0;
+      SYNCED = 1;                // node is synced
+      SYNCING = 2;               // node is syncing (but not synced)
+      NOT_SYNCING = 3;           // sync did not start
+      SYNC_FAILED = 4;           // node sync fail event. Node will never be syncing or synced again in this session.
+      NEIGHBOR_CONNECTED = 5;    // Connected to a neighbor
```

Comment:
Is this useful by itself? E.g., without the node id of the new neighbor?

Reply:
probably not very useful. I was trying to get away from adding a separate network status events streams. So the idea is that when such a status event is received, node client can query the node state for number of neighbors currently connected to figure out the net status. This is just basically a hint that p2p connectivity status changed. Maybe not the  best way to do it. Maybe it is better to just have NET_CHANGED event and clients may decide to query the node for additional net info when they get such an event from a node. Again - instead of having to constantly poll...

Reply:
Okay